nityam_kumar/PostgresAssignment/src/controller/mark.ts

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { Context } from "vm";
+import * as marksService from "../services/mark";
+
+import markSchema from "../db/validateSchema/markSchema";
+import AppError from "../utils/appError";
+
+import {
+  classNoSchema,
+  marksInputSchema,
+  fetchTopBYNumberSchema,
+  subjectIDSchema,
+  studentIDSchema,
+} from "../db/validateSchema/helperSchema";
+
+interface IMark {
+  student_id: string;
+  subject_id: string;
+  marks: number;
+  teacher_id: string;
+  class_number: number;
+}
+
+export const createMarks = async (ctx: Context) => {
+  const m1: IMark = ctx.request.body;
+
+  try {
+    await markSchema.validateAsync(m1);
+  } catch (err) {
+    throw new AppError(err.message, 400);
+  }
+
+  await Promise.all([
```

Comment:
what happens if any of the function returns true ???

Reply:
marksService.checkStudentExist(m1),
    marksService.checkSubjectExist(m1),
    marksService.checkAlreadyExist(m1),

before creating a new entry in marks table it will check for existence if student,subject and if entry is already available in marks table and after passing all check  marks entry will be created

Reply:
code changed